### DISCUSSION TITLE
Meter 1 Code Sensor

### DISCUSSION THREAD

**Discussion by paulrumsby** [nathanmarlor/foxess_modbus#369] [Ideas]
Hi Nathan

Is it possible to add a sensor that reads the value of the meter 1 code, which I believe is register 40010 on the H1 inverter? The sensor could be used to write an automation that will trigger if the meter 1 code becomes disabled, as it did recently for me with a CT data logger firmware update, and reset it to 1=CT. I can not see anywhere on the fox cloud site where this can be done and I had to ask the installer to re-enable it for me. 

Thanks for reading

Paul

**Comment by FozzieUK** [comment]
Hi Paul, 
The CT meter being disabled is unfortunately a fairly common fault after a long period of the inverter being shut down or a firmware upgrade, but, the better news is that it is a one time event after first installation i.e. once it has happened, it doesn't appear to occur again - perhaps a problem with earlier firmware versions but i’ve never seen it happen again on any system. 
If you have the inverter passcode (usually 0,0,0,0) you can see/enable it from the front panel in settings, or if you have an installer/agent account the app will allow you to configure it.

I’ll leave it to Antony / Nathan to consider adding to the list for new sensors, albeit one that would probably be disabled by default - the issue being the number of registers being polled and the impact that has on some of the slower RS485 devices (W610 being the most prevalent in use and the worst candidate).

**Reply by paulrumsby** [reply]
Hi Dave,

Thanks for the response and information, that is goof news if it is a one time event then I may have been lucky and got it out of the way early on. Only had the system installed 5 weeks ago. Thank again...

Paul

**Comment by canton7** [comment]
> as it did recently for me with a CT data logger firmware update, and reset it to 1=CT.

Normally what happens is that it gets reset to Disabled, rather than being switched to CT. I was going to say, it might be worth monitoring this register to and raising an error if CT becomes disabled (as you say, this is a common problem). However in your case it sounds like this didn't happen, in which case this wouldn't help you.

If we do implement raising an error if CT is disabled then this entity would have to be enabled by default. That does affect people using the W610 over TCP, but I did recently have to increase the poll period to 15 seconds there (polling the error registers pushed us over 10s), which gives us a bit of headroom to add some more registers.
